# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CodeGenerator.cpp
# Commit: b24505537d70
# Full Hash: b24505537d700cd4e01adee576645a617b6ef935
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 1: Reserve two slots after stack arguments for the future tls preservation. r=lth
#   
#   
#   We are going to remove Frame::tls and support trampolines for indirect calls, so we need to get rid of using Frame::tls.
#   In this and the followup patches I will iteratively remove all dependencies of Frame::tls and remove it eventually.
# ==============================================================================

diff -r 653e90061f03 -r b24505537d70 js/src/jit/CodeGenerator.cpp
--- a/js/src/jit/CodeGenerator.cpp	Fri Oct 30 08:31:42 2020 +0000
+++ b/js/src/jit/CodeGenerator.cpp	Fri Oct 23 18:17:36 2020 +0000
@@ -15323,7 +15323,7 @@
   const wasm::FuncExport& funcExport = lir->mir()->funcExport();
   const wasm::FuncType& sig = funcExport.funcType();
 
-  ABIArgGenerator abi;
+  WasmABIArgGenerator abi;
   for (size_t i = 0; i < lir->numOperands(); i++) {
     MIRType argMir;
     switch (sig.args()[i].kind()) {